pkg/common/pod.go

```diff
@@ -40,3 +43,29 @@ func GetAllPodsWithLabel(r ReconcilerCommon, labelSelectorMap map[string]string,
 
 	return podList, nil
 }
+
+// DeletePodsWithLabel - Delete all pods in namespace of the obj matching label selector
+func DeletePodsWithLabel(r ReconcilerCommon, obj metav1.Object, labelSelectorMap map[string]string) error {
+	// delete pods
+	podList := &corev1.PodList{}
+	listOpts := []client.ListOption{
+		client.InNamespace(obj.GetNamespace()),
+		client.MatchingLabels(labelSelectorMap),
+	}
+
+	if err := r.GetClient().List(context.Background(), podList, listOpts...); err != nil {
+		err = fmt.Errorf("Error listing pods for %s: %v", obj.GetName(), err)
+		return err
+	}
+
+	// delete all pods
+	for _, pod := range podList.Items {
+		err := r.GetClient().Delete(context.TODO(), &pod)
+		if err != nil && !k8s_errors.IsNotFound(err) {
+			err = fmt.Errorf("Error deleting pod %s: %v", pod.Name, err)
+			return err
+		}
+	}
+
+	return nil
```

Comment:
Just curious here: could a `r.Client.DeleteAllOf()` do the trick here without needing to get the list and then delete one-by-one?  I've never used the function myself.  Fine either way.

Reply:
oh nice, wasn't aware of this. will check it!

Reply:
done